### ISSUE TITLE
[YSQL] Secondary index read does not use a consistent read point.

### ISSUE THREAD

**Issue by robertpang** [YugaByte/yugabyte-db#990]
In `YBCIndexExecuteSelect()`, the YBC select statement to fetch the row from the base table using the `ybctid` is not created with a read point:

```
HeapTuple
YBCIndexExecuteSelect(Relation relation, Datum ybctid)
{
    YBCPgStatement ybc_stmt;
    TupleDesc      tupdesc = RelationGetDescr(relation);

    HandleYBStatus(YBCPgNewSelect(ybc_pg_session,
                                  YBCGetDatabaseOid(relation),
                                  RelationGetRelid(relation),
                                  InvalidOid,
                                  &ybc_stmt,
                                  NULL /* read_time */));
...
}
```

Without the read point, the index select may not be reading the rows from the base table with a consistent read point.

**Comment by nocaway** [comment]
@ndeodhar This has been fixed.

**Comment by kmuthukk** [comment]
Fixed as part of #1894 and #1611
